src/stream/src/executor/source/fetch_executor.rs

```diff
@@ -297,6 +332,30 @@ impl<S: StateStore> FsFetchExecutor<S> {
                                         .set(split_id, fs_split.encode_to_json())
                                         .await?;
                                 }
+                                }
+                                else if let risingwave_connector::source::ConnectorProperties::Gcs(_)=properties{
+                                    let fs_split: OpendalFsSplit<GcsProperties> =
+                                    match row.datum_at(1) {
+                                        Some(ScalarRefImpl::Jsonb(jsonb_ref)) => {
+                                            OpendalFsSplit::restore_from_json(
+                                                jsonb_ref.to_owned_scalar(),
+                                            )?
+                                        }
+                                        _ => unreachable!(),
+                                    };
+
+                                if offset.parse::<usize>().unwrap() >= fs_split.size {
+                                    splits_on_fetch -= 1;
+                                    state_store_handler.delete(split_id).await?;
```

Comment:
How do we make sure we won't re-fetch a already fetched file if we delete it from the state table when we finish fetching the whole file? I thought lister may maintain some states for already listed files if we do this in fetcher but I didn't find relevant codes. Sorry for missing this in the PR when we first introduce the list and fetch executor, I should have asked this question then.

Reply:
The plan here is like
```
                                  |--> dedup ---> fetcher
lister(singleton) ---> exchange --|--> dedup ---> fetcher
                                 ...

```

Reply:
Discussed with @tabVersion offline, there is a dedup executor sitting in between lister and fetcher to dedup file name.

Reply:
@wcy-fdu please add the discussion part into the pr description.